src/Package/Impl/Shell/VsSettingsStorage.cs

```diff
@@ -67,21 +85,21 @@ internal sealed class VsSettingsStorage : ISettingsStorage {
 
         public void SetSetting(string name, object value) {
             lock (_lock) {
-                _settings[name] = value;
+                _settingsCache[name] = value;
             }
         }
 
         public void Persist() {
             lock (_lock) {
-                foreach (var s in _settings) {
+                foreach (var s in _settingsCache) {
                     var t = s.Value.GetType();
-                    if (typeof(bool).IsAssignableFrom(t)) {
+                    if (s.Value is bool) {
                         _store.SetBoolean(_collectionPath, s.Key, (bool)s.Value);
                     } else if (typeof(int).IsAssignableFrom(t) || t.IsEnum) {
```

Comment:
I guess since compiler doesnt like such assignments either. :-) Will add separate clause in master.